### PR TITLE
Fix bugs when encryption flow is enabled in the API

### PR DESCRIPTION
**Pull request by holachuy**
This is a collection of bug fixes to complete the implementation of the encryption flow (using encrypted auth credentials in cookies and storing metadata state in storage) for the API server. #3

### CODE REVIEW

File: config/environments/local/settings/common.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 env: LOCAL
 cloud: LOCAL
+encryptedFlow: true
```
Comment (seehamrun) [inline config/environments/local/settings/common.yaml]: Just a note, for the encrypted flow, we need to bring up the worker binary separately when testing locally right? Is there already documentation around that?

File: portability-api/src/main/java/org/dataportabilityproject/webapp/CryptoHelper.java [hunk 2]
```diff
@@ -16,14 +16,15 @@
 package org.dataportabilityproject.webapp;
 
 import static org.apache.axis.transport.http.HTTPConstants.HEADER_SET_COOKIE;
+import static org.apache.axis.transport.http.HTTPConstants.HEADER_SET_COOKIE2;
```
Comment (seehamrun) [inline portability-api/src/main/java/org/dataportabilityproject/webapp/CryptoHelper.java]: remove

File: portability-core/src/test/java/org/dataportabilityproject/job/JobDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package org.dataportabilityproject.job;
```
Comment (seehamrun) [inline portability-core/src/test/java/org/dataportabilityproject/job/JobDaoTest.java]: Nit: license header missing

File: portability-worker/src/main/resources/settings/common.yaml [hunk 4]
```diff
@@ -1,2 +1,11 @@
 env: LOCAL
```
Comment (seehamrun) [inline portability-worker/src/main/resources/settings/common.yaml]: Isnt this file copied over by the build script?

File: portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java [hunk 5]
```diff
@@ -182,12 +175,18 @@ public void updateJobStateToAssigneWithAuthData(String id, String encryptedExpor
...
 
-    updateJob(updatedJob, JobState.ASSIGNED_WITH_AUTH_DATA);
-    // Delete the job in the previous state
-    String previousKey = createKey(JobState.ASSIGNED_WITHOUT_AUTH_DATA, id);
-    storage.delete(
-        previousKey); // TODO: Determine whether to Tombstone or Delete from unassigned once it's entered in ready state
+  /**
+   * Deletes an existing {@code job} in storage with the given {@code jobState}.
+   */
+  public void deleteJob(String id, JobState jobState) throws IOException {
```
Comment (seehamrun) [inline portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java]: It looks like deleteJob is only called in testing - is this something that we can expose for TestOnly?

File: portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java [hunk 6]
```diff
@@ -39,7 +39,7 @@
    * The value ASSIGNED_WITHOUT_AUTH_DATA indicates the client has submitted all data required, such as the
    * encrypted auth data, in order to begin processing the portability job.
    */
-  enum JobState {
+  public enum JobState {
```
Comment (seehamrun) [inline portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java]: This also looks like (outside of this class) its only used in Test, is there a way we can limit the visibility?

File: config/environments/local/settings/common.yaml (hunk 1 above)
Comment (holachuy) [reply config/environments/local/settings/common.yaml]: Reverted. This is not ready yet.

File: portability-api/src/main/java/org/dataportabilityproject/webapp/CryptoHelper.java (hunk 2 above)
Comment (holachuy) [reply portability-api/src/main/java/org/dataportabilityproject/webapp/CryptoHelper.java]: Done

File: portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java (hunk 6 above)
Comment (holachuy) [reply portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java]: Annotated with @VisibleForTesting

File: portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java (hunk 5 above)
Comment (holachuy) [reply portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java]: I think we'll eventually need it when implementing the restart/recovery parts of the lifecycle

File: portability-core/src/test/java/org/dataportabilityproject/job/JobDaoTest.java (hunk 3 above)
Comment (holachuy) [reply portability-core/src/test/java/org/dataportabilityproject/job/JobDaoTest.java]: Done

File: portability-worker/src/main/resources/settings/common.yaml (hunk 4 above)
Comment (holachuy) [reply portability-worker/src/main/resources/settings/common.yaml]: Let me add a .gitignore

**Comment by holachuy** [comment]
Thanks for the review. I think I addressed all the comments. PTAL.

File: portability-worker/src/main/resources/settings/common.yaml (hunk 4 above)
Comment (seehamrun) [reply portability-worker/src/main/resources/settings/common.yaml]: This file still shows up in the diff of the request

File: portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java (hunk 5 above)
Comment (seehamrun) [reply portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java]: Oh right, that makes sense! Thanks!

**Pull request merged**